### PR TITLE
Simplifies chart rendering

### PR DESCRIPTION
**Pull request by wesalvaro**
Removes promises and relies more on events and property computation.

Adds `drawn` property to chart.
Changes `drawChart()` to the more apt `redraw()`.

### CODE REVIEW

**Comment by wesalvaro** [comment]
@ebidel There are some extra commits in here that need to be cleaned up, but if you would take a look, that would be great.

File: google-chart.html [hunk 1]
```diff
@@ -214,65 +214,62 @@
       selection: {
         type: Array,
         notify: true,
-        observer: '_setSelection'
+        observer: '_setSelection',
```
Comment (ebidel) [inline google-chart.html]: remove the trailing "," on these

File: google-chart.html [hunk 2]
```diff
@@ -214,65 +214,62 @@
...
+      },
+
+      drawn: {
+        type: Boolean,
+        readOnly: true,
+        reflectToAttribute: true,
```
Comment (ebidel) [inline google-chart.html]: does this need to reflect?

File: google-chart.html [hunk 3]
```diff
@@ -306,29 +301,35 @@
      * @return {string} Returns image URI.
      */
     get imageURI() {
-      return this._chartObject.getImageURI();
+      if (!this._chart) return;
```
Comment (ebidel) [inline google-chart.html]: brackets

File: google-chart.html [hunk 4]
```diff
@@ -306,29 +301,35 @@
...
+      return this._chart.getImageURI();
     },
 
     // Handles changes to the DataView object attribute.
     _viewChanged: function() {
       if (!this.view) return;
```
Comment (ebidel) [inline google-chart.html]: can you add brackets here

File: google-chart.html [hunk 5]
```diff
@@ -306,29 +301,35 @@
...
-      this._dataView = Promise.resolve(this.view);
-      this.drawChart();
+      this._dataView = this.view;
     },
 
     // Handles changes to the rows & columns attributes.
     _rowsOrColumnsChanged: function() {
       var rows = this.rows, cols = this.cols;
       if (!rows || !cols) return;
```
Comment (ebidel) [inline google-chart.html]: can you add brackets here

**Comment by wesalvaro** [comment]
Alll comments addressed and extra commits removed.

**Comment by wesalvaro** [comment]
Going to fix up the Closure Compiler issues before merging.
Looks like it was half-fixed internally, out-of-phase with github... ='(

**Comment by wesalvaro** [comment]
OK. Added the commit to fix #110 
Should I fixup that commit or leave it separate?
I'm good to go with this PR and with release 1.1.0

**Comment by wesalvaro** [comment]
Aw, dang it. I somehow jacked it up...

**Pull request merged**